asmcomp/arm64/emit.mlp

```diff
@@ -1009,6 +1032,30 @@ let rec emit_all env i =
 
 (* Emission of a function declaration *)
 
+type preproc_fun_result =
+  { max_stack_size : int;
+    contains_nontail_calls : bool }
+
+let preproc_fun env fun_body _fun_name =
+  let rec proc_instr r s i =
+    if i.desc = Lend then r else
+        let upd_size r delta =
+          {r with max_stack_size = max r.max_stack_size (s+delta)}
+        in
+        let (r',s') = match i.desc with
+          | Lop (Istackoffset n) -> (upd_size r n, s+n)
+          | Lpushtrap _ -> (upd_size r 16, s+16)
+          | Lpoptrap -> (r, s-16)
+          | Lop (Icall_ind | Icall_imm _ ) ->
+              ({r with contains_nontail_calls = true}, s)
+          | _ -> (r, s)
+        in
+        proc_instr r' s' i.next
+  in
+  let fs = frame_size env in
+  let r = {max_stack_size = fs; contains_nontail_calls = false} in
```

Comment:
`s/max_stack_size/max_frame_size`. Also affects `amd64/emit.mlp`. 

Perhaps we should move this function to `asmcomp/emitaux.ml` instead of duplicating this for every backend.